src/backend/cdb/motion/ic_udpifc.c

```diff
@@ -1167,6 +1169,105 @@ resetRxThreadError()
 }
 
 
+/*
+ * isIPv6Available
+ * 		Check whether a valid IPv6 address exists in the system.
+ *
+ * 		We want to have both loopback IPv6 address, which will show if IPv6 was
+ * 		disabled earlier, and some non-link-local address currently available
+ * 		for our use.
+ */
+static bool
+isIPv6Available(void)
+{
+	bool		hasIpv6Addr = false;
+	bool		hasIpv6Loopback = false;
+	int			errnoSave = errno;
+
+	int			errorErrnoSave = 0;
+	char	   *errorCause = NULL;
+
+	struct ifaddrs	*ifaddr = NULL;
+	struct ifaddrs	*ifaddrFirst = NULL;
+
+	if (getifaddrs(&ifaddrFirst) == -1)
+	{
+		errorCause = "getifaddrs()";
+		goto end;
+	}
+
+	for (ifaddr = ifaddrFirst; ifaddr != NULL; ifaddr = ifaddr->ifa_next)
+	{
+		if (ifaddr->ifa_addr != NULL &&
+			ifaddr->ifa_addr->sa_family == AF_INET6)
+		{
+			const struct sockaddr_in6 *addr =
+				(struct sockaddr_in6 *) ifaddr->ifa_addr;
+			const struct in6_addr *sinAddr = &addr->sin6_addr;
+
+			/* Discard wildcard, and link-local addresses. */
+			if (!IN6_IS_ADDR_UNSPECIFIED(sinAddr) &&
+				!IN6_IS_ADDR_LINKLOCAL(sinAddr))
+			{
+				/* Make sure the address can be used in bind(). */
+				int			bindFd = -1;
```

Comment:
```suggestion
				int			bindFd;
```
or
```suggestion
				int			bindFd = socket(AF_INET6, SOCK_DGRAM, 0);
```

Reply:
We don't want to create a socket if the iteration can be skipped by very next if condition. What's wrong with initializing the descriptor to -1?

Reply:
> What's wrong with initializing the descriptor to -1?

it is redundant

Reply:
> it is redundant

I think it does not really matter.